kvcache/causal.go

```diff
@@ -681,3 +681,36 @@ func (c *Causal) Remove(seq int, beginIndex, endIndex int32) error {
 
 	return nil
 }
+
+func (c *Causal) Has(seq int, pos int32) bool {
```

Comment:
I'm a little unsure about two things here:

1. How does context shifting affect this, if at all?
2. For the SWA check on line 703 below, is checking the highest position correct?